### ISSUE TITLE
Making custom "Type1" dataset for training

### ISSUE THREAD

**Issue by unleft** [clovaai/spade#20]
Thank you for sharing this amazing work. After following the readme file and seeing that the SPADE seems to work well (despite some hiccups to which I managed to find temporary workarounds), I decided to try to train and test SPADE on a custom dataset. 

I am trying to replicate the format of _Type 1_ data. I edited the field representers on the config files, but I am stuck on how I should define and set the **_"labels"._** The readme file says that it's a pair of "(n_field + n_text) x n_text adjacency matrix expressing serialization/grouping", but I don't think I'm following well on how I should set the _values_.

Could you please explain how they're defined, or how they are made?

**Comment by whwang299** [comment]
Hi @unleft 

Sorry about the hiccups 😢. 
About the label, you may refer to this part.
https://github.com/clovaai/spade/blob/a85574ceaa00f1878a23754f283aa66bc2daf082/spade/model/data_utils.py#L212

Briefly, it is an adjacency matrix (let's call it M) with (n_field + n_text) rows and (n_text) columns.
For example, if M_{ij} = 1, there is a directed arrow from i-th token to j-th token.
If it is about the serialization, it means i-th token is followed by j-th token. 
If it is about the grouping, j-th token is a first token of the sub-group.

Wonseok

**Comment by unleft** [comment]
Thanks for the quick reply. After a bit of analysing the input data, I think I am finally starting to get the overall idea of how the labels are defined.

Few more questions, though. Is there any default value set for paths to which the trained model/weights are saved? 
I want to periodically retrain the model to keep it up to date, as well as have it recognize other languages such as Korean and Japanese more efficiently. I'd like to know if I can use the pretrained model file "model.pt" to update/fine-tune it :)

**Comment by whwang299** [comment]
For the save path, please refer to the following lines.
https://github.com/clovaai/spade/blob/a85574ceaa00f1878a23754f283aa66bc2daf082/spade/model/model.py#L466-L468

To load pre-trained models, see 
https://github.com/clovaai/spade/blob/a85574ceaa00f1878a23754f283aa66bc2daf082/spade/model/run_model.py#L40-L47

By the way, there is a [useful callback from `pytorch-lightning`](https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.callbacks.model_checkpoint.html).
So if you want to futher customize the script, you may refer to the link.

Wonseok.

**Comment by unleft** [comment]
Thank you so much for the detailed instruction!
